### PR TITLE
Settings overhaul 2

### PR DESCRIPTION
**Pull request by orangewolf**
Ref #1719 

Most of the credit for this work goes to Ubiquity Press. Many of the lines here are extracted from https://github.com/ubiquitypress/hyku_addons

This development was funded by the British Library: https://bl.iro.bl.uk

Part 2 of settings overhaul #1746 . This work is focuses on making all the settings either functional or marking them as disabled.  This was done to keep compatibility with the existing Ubiquity data layout. 

This PR will focus on the follwing
- Port or discard implementations of existing settings
- Align all existing settings and remove the Settings class
- Set defaults based on Hyrax env var names

Admin interface will be a seperate PR.

@samvera/hyku-code-reviewers

### CODE REVIEW

File: app/models/concerns/account_settings.rb [hunk 1]
```diff
@@ -12,26 +15,33 @@ module AccountSettings
     end
 
     setting :allow_signup, type: 'boolean', default: true
-    setting :bulkrax_validations, type: 'boolean'
+    setting :bulkrax_validations, type: 'boolean', disabled: true
```
Comment (orangewolf) [inline app/models/concerns/account_settings.rb:18]: disabled here doesn't _do_ anything yet. it just identifies those setting that Hyku doesn't use at the moment. Most of these are things that are in hyku_addons or other places but haven't made it in to hyku yet.  I'm designating them so I can skip them in the UI until they are doing something.

File: README.md [hunk 2]
```diff
@@ -125,9 +125,9 @@ Hyku relies on the helm charts provided by Hyrax. See [Deployment Info](https://
 
 Much of the default configuration in Hyku is set up to use multi-tenant mode.  This default mode allows Hyku users to run the equivielent of multiple Hyrax installs on a single set of resources. However, sometimes the subdomain splitting multi-headed complexity is simply not needed.  If this is the case, then single tenant mode is for you.  Single tenant mode will not show the tenant sign up page, or any of the tenant management screens. Instead it shows a single Samvera instance at what ever domain is pointed at the application.
 
-To enable single tenant, in your settings.yml file change multitenancy/enabled to `false` or set `SETTINGS__MULTITENANCY__ENABLED=false` in your `docker-compose.yml` and `docker-compose.production.yml` configs. After changinig this setting, run `rails db:seed` to prepopulate the single tenant.
+To enable single tenant, set `HYKU_MULTITANT=false` in your `docker-compose.yml` and `docker-compose.production.yml` configs. After changinig this setting, run `rails db:seed` to prepopulate the single tenant.
```
Comment (chauncey-garrett) [inline README.md:128]: `HYKU_MULTITENANT`

(spelling)

**Review by chauncey-garrett — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (orangewolf) [reply README.md]: thank you!

**Comment by orangewolf** [comment]
moved to #1771

**Pull request closed without merging**